custom_components/tuya_local/devices/yym_805SW_aroma_nightlight.yaml

```diff
@@ -7,18 +8,114 @@ primary_entity:
     - id: 1
       name: switch
       type: boolean
+      mapping:
+        - dps_val: true
+          icon: mdi:scent
+          value: ON
+        - dps_val: false
+          icon: mdi:scent-off
+          value: OFF      
     - id: 2
       name: speed
       type: string
+      hidden: true
       mapping:
         - dps_val: large
           value: 100
+          icon: "mdi:sprinkler-variant"
         - dps_val: small
           value: 50
+          icon: "mdi:sprinkler"
+    - id: 4          
```

Comment:
If this is related to the Timer, you should put it on that entity, or make a separate sensor entity for it.

Reply:
Moved to as attribute timer. Update is not very frequent, and even reports "nullmin" sometimes, so I don't find it useful enough to be separate sensor.